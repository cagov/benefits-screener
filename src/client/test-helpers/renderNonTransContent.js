import React from "react";
import { shallow } from "enzyme";
import { I18nextProvider, Trans } from "react-i18next";
import i18n from "./i18nForTests";
import shallowUntilTarget from "./shallowUntilTarget";

function renderNonTransContent(Component, props = {}) {
  const providersWrapper = shallow(
    <I18nextProvider i18n={i18n}>
      <Component {...props} />
    </I18nextProvider>
  );

  const wrapper = shallowUntilTarget(providersWrapper, Component.name);
  // Incorrectly using renderNonTransContent to render a component that
  // contains a Trans component results in a silent failure (the rendered
  // snapshot will contain the Trans component itself instead of the rendered
  // contents). This test makes that silent failure more obvious.
  expect(wrapper.containsMatchingElement(<Trans />)).toBe(false);

  return wrapper;
}

module.exports = renderNonTransContent;
