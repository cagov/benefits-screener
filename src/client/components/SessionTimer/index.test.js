import renderNonTransContent from "../../test-helpers/renderNonTransContent";
import Component from "./index";

describe("<SessionTimer />", () => {
  it("startOrUpdate", async () => {
    const wrapper = renderNonTransContent(Component, {
      action: "startOrUpdate",
      setUserData: () => {},
    });

    expect(wrapper).toMatchSnapshot();
    expect(Component.getTimerIdForTest()).not.toBeNull();
    clearTimeout(Component.getTimerIdForTest());
  });

  it("clear", async () => {
    const wrapper = renderNonTransContent(Component, {
      action: "clear",
      setUserData: () => {},
    });

    expect(wrapper).toMatchSnapshot();
    expect(Component.getTimerIdForTest()).toBeNull();
  });
});
