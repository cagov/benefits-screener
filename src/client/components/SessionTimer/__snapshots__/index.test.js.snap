// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SessionTimer /> clear 1`] = `
<Bootstrap(Modal)
  centered={true}
  onHide={[Function]}
>
  <ModalHeader
    className="border-0"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle>
      <strong>
        Your Session Will End Soon
      </strong>
    </ModalTitle>
  </ModalHeader>
  <ModalBody>
    <Trans
      i18nKey="timeout-modal.warning"
      t={[Function]}
      values={
        Object {
          "numberOfMinutes": 5,
        }
      }
    />
  </ModalBody>
  <ModalFooter
    className="border-0"
  >
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="secondary"
    >
      Continue Working
    </Button>
  </ModalFooter>
</Bootstrap(Modal)>
`;

exports[`<SessionTimer /> startOrUpdate 1`] = `
<Bootstrap(Modal)
  centered={true}
  onHide={[Function]}
>
  <ModalHeader
    className="border-0"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle>
      <strong>
        Your Session Will End Soon
      </strong>
    </ModalTitle>
  </ModalHeader>
  <ModalBody>
    <Trans
      i18nKey="timeout-modal.warning"
      t={[Function]}
      values={
        Object {
          "numberOfMinutes": 5,
        }
      }
    />
  </ModalBody>
  <ModalFooter
    className="border-0"
  >
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="secondary"
    >
      Continue Working
    </Button>
  </ModalFooter>
</Bootstrap(Modal)>
`;
